### PR TITLE
pymol: init at 1.8.4

### PR DESCRIPTION
**Pull request by Mounium**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mounium** [comment]
Any idea on why have it failed?

File: pkgs/applications/science/chemistry/pymol/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl, python27Packages, glew, freeglut, libpng, libxml2, tk, freetype }:
+
+let
+  version = "1.8.4.0";
+in
+stdenv.mkDerivation {
```
Comment (fpletz) [inline pkgs/applications/science/chemistry/pymol/default.nix]: There is also `buildPythonApplication` in `pythonPackages` which should automate all the phases you have overriden. Have you tried that?

File: pkgs/applications/science/chemistry/pymol/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    python setup.py build
+  '';
+  
+  installPhase = ''
+    python setup.py install --home=$out
+  '';
+
+  meta = {
+    description = "Molecule editor and visualizer";
+  };
```
Comment (fpletz) [inline pkgs/applications/science/chemistry/pymol/default.nix]: Please include more information in the meta section. At least the license and the homepage should be present.

**Review by fpletz — CHANGES_REQUESTED** [review]

**Comment by Mounium** [comment]
@fpletz Applied buildPhytonApplication, and added license as well as a bit more detailed description as well as the website. Although I still doubt that it will build on Mac.

Note: This version of PyMol does not utilises Python megawidgets, as I could not find this package on nixpkgs. It doesn't cause any problem:

>  7. Python megawidgets (Pmw -- only required for the "external" GUI)

File: pkgs/applications/science/chemistry/pymol/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  version = "1.8.4.0";
+in
+python27Packages.buildPythonPackage {
+  name = "pymol-${version}";
+  src = fetchurl {
+    url = "mirror://sourceforge/project/pymol/pymol/1.8/pymol-v1.8.4.0.tar.bz2";
+    sha256 = "0yfj8g5yic9zz6f0bw2n8h6ifvgsn8qvhq84alixsi28wzppn55n";
+  };
+
+  buildInputs = [ python27Packages.numpy glew freeglut libpng libxml2.dev tk freetype ];
```
Comment (Mic92) [inline pkgs/applications/science/chemistry/pymol/default.nix]: Adding `libxml2` to `buildInputs` should automatically pull `libxml2.dev` as well.

File: pkgs/applications/science/chemistry/pymol/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchurl, python27Packages, glew, freeglut, libpng, libxml2, tk, freetype }:
+
+let
+  version = "1.8.4.0";
+in
+python27Packages.buildPythonPackage {
```
Comment (Mic92) [inline pkgs/applications/science/chemistry/pymol/default.nix]: `python27Packages.buildPythonApplication` is probably wanted here.

**Pull request merged**